core/toolmanager.cpp

```diff
@@ -136,11 +136,8 @@ ToolData ToolManager::toolInfoForFactory(ToolFactory *factory) const
 
 bool ToolManager::hasTool(const QString &id) const
 {
-    for (ToolFactory *factory : m_tools) {
-        if (factory->id() == id)
-            return true;
-    }
-    return false;
+    return std::any_of(m_tools.begin(), m_tools.end(),
```

Comment:
Learned a new hammer, and now everything is a nail? :)

While technically correct, the use of any_of to mean find the _one_ item that matches is nonsensical.